### ISSUE TITLE
libonnxruntime_providers_dnnl.so not installed by CMake

### ISSUE THREAD

**Issue by xkszltl** [microsoft/onnxruntime#4729]
**Describe the bug**

`libonnxruntime_providers_dnnl.so` not installed in pure cmake workflow.
```
cmake ....
cmake --build . --target install
```
It's only installed for pip.
This may be a mistake since DNNL provider should have nothing to do with python.

**Urgency**

We have user blocked by this due to missing `.so` when running ort.
Mitigation with `LD_LIBRARY_PATH` to site-packages is provided but it's not a good solution. 

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): CentOS 7/Ubuntu 18.04
- ONNX Runtime installed from (source or binary): source
- ONNX Runtime version: 1.4.0
- Python version: 3.6
- GCC/Compiler version (if compiling from source): gcc-8

**To Reproduce**

Basically
`cmake -Donnxruntime_BUILD_SHARED_LIB=ON -Donnxruntime_USE_DNNL=ON -DCMAKE_BUILD_TYPE=Release -GNinja ../cmake && cmake --build . --target install`
Complete script in here:
https://github.com/xkszltl/Roaster/blob/273467b574042f357a450ced4f20587de0c66acc/pkgs/ort.sh#L71-L111

**Comment by pranavsharma** [comment]
cc @RyanUnderhill

**Comment by xkszltl** [comment]
Any update?

**Comment by xkszltl** [comment]
@RyanUnderhill @pranavsharma Any update?

**Comment by jywu-msft** [comment]
that issue should be fixed by https://github.com/microsoft/onnxruntime/commit/16d35266ab9e060c5866a0cb629e37b50015130e#diff-543ceb9d78511f8f6dc93a993e131d59bccaa53bf82d8a02f268ca32901201ca

**Comment by xkszltl** [comment]
Verified

**Comment by SuganPrabu96** [comment]
@jywu-msft Hi! Thank you for fixing this. I was facing the same issue as well. Is there a plan for releasing this as a patch? For now, I was considering checking out one of the release versions and cherry picking this commit but in case you are releasing soon, I thought I could just use the new release version instead.
